src/commands/api/deployImageApi.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { callWithMaskHandling, callWithTelemetryAndErrorHandling, createSubscriptionContext, type ExecuteActivityContext, type IActionContext, type ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { ImageSource, acrDomain } from "../../constants";
+import { type DeployImageApiTelemetryProps as TelemetryProps } from "../../telemetry/commandTelemetryProps";
+import { type SetTelemetryProps } from "../../telemetry/SetTelemetryProps";
+import { getDomainFromRegistryName, getRegistryFromAcrName } from "../../utils/imageNameUtils";
+import { pickContainer } from "../../utils/pickItem/pickContainer";
+import { deployImage } from "../deployImage/deployImage";
+import { type ContainerRegistryImageSourceContext } from "../image/imageSource/containerRegistry/ContainerRegistryImageSourceContext";
+import { type ImageSourceBaseContext } from "../image/imageSource/ImageSourceContext";
+import { type DeployImageToAcaOptionsContract } from "./vscode-azurecontainerapps.api";
+
+export type DeployImageApiContext = ImageSourceBaseContext & ExecuteActivityContext & SetTelemetryProps<TelemetryProps>;
+
+export async function deployImageApi(deployImageOptions: DeployImageToAcaOptionsContract): Promise<void> {

```

Comment:
This function is all carryover code except with a new `callWithTelemetryAndErrorHandling` wrapper